binderhub/builder.py

```diff
@@ -408,31 +408,69 @@ async def get(self, provider_prefix, _unescaped_spec):
             else:
                 image_found = True
 
+        require_build_only = self.settings.get("require_build_only", False)
+        build_only = False
+        if not require_build_only:
+            build_only_query_parameter = self.get_query_argument(
+                name="build_only", default=""
+            )
+            if build_only_query_parameter.lower() == "true":
+                raise HTTPError(
+                    log_message="Building but not launching is not permitted!"
```

Comment:
The error stems from not being passed a `build_only=true` query parameter when BinderHub is configured with `require_build_only`. I think it would be good to surface that more explicitly as well in this message to help anyone seeing this directly or indirectly via a issue report figure out what is going on.

The error message can be lengthy without issues, so we can include both a non-technical description as well.